File: archive/ios-swiftui/RedactPDF/Models/RedactionModels.swift
import CoreGraphics
import Foundation
import SwiftUI

enum RedactionMode: String, CaseIterable, Identifiable {
    case smart = "AI Detect"
    case manual = "Manual Draw"

    var id: String { rawValue }
}

enum RedactionStyle: String, CaseIterable, Identifiable {
    case black = "Solid Black"
    case blur = "Gaussian Blur"
    case white = "Solid White"

    var id: String { rawValue }

    var previewColor: Color {
        switch self {
        case .black: return .black
        case .blur: return Color.gray.opacity(0.6)
        case .white: return .white
        }
    }
}

struct RedactionRect: Identifiable, Equatable {
    let id: UUID
    var pageIndex: Int
    var rect: CGRect
    var style: RedactionStyle

    init(id: UUID = UUID(), pageIndex: Int, rect: CGRect, style: RedactionStyle = .black) {
        self.id = id
        self.pageIndex = pageIndex
        self.rect = rect
        self.style = style
    }
}

struct LoadedPDFDocument: Identifiable, Equatable {
    let id = UUID()
    let url: URL
    let title: String

    static func == (lhs: LoadedPDFDocument, rhs: LoadedPDFDocument) -> Bool {
        lhs.id == rhs.id
    }
}

File: archive/flutter/lib/models/redaction_models.dart
import 'package:flutter/material.dart';
import 'package:uuid/uuid.dart';

enum RedactionMode {
  smart('AI Detect'),
  manual('Manual Draw');

  const RedactionMode(this.label);
  final String label;
}

enum RedactionStyle {
  black('Solid Black', Colors.black),
  blur('Gaussian Blur', Color(0x99FFFFFF)),
  white('Solid White', Colors.white);

  const RedactionStyle(this.label, this.previewColor);
  final String label;
  final Color previewColor;
}

class RedactionRect {
  RedactionRect({
    String? id,
    required this.pageIndex,
    required this.rect,
    this.style = RedactionStyle.black,
  }) : id = id ?? const Uuid().v4();

  final String id;
  final int pageIndex;
  final Rect rect;
  final RedactionStyle style;
}

class LoadedPdfDocument {
  LoadedPdfDocument({
    required this.path,
    required this.title,
  });

  final String path;
  final String title;
}
